gammapy/background/off_data_background_maker.py

```diff
@@ -160,38 +164,55 @@ def make_model(self, modeltype):
             # Build the model
             if modeltype == "3D":
                 model = CubeBackgroundModel.define_cube_binning(obs_table_group, method='default')
-                # TODO: should adapt to pass `obs_ids` here like for 2D below
                 model.fill_obs(obs_table_group, self.data_store)
                 model.smooth()
                 model.compute_rate()
-                self.models3D.append(model)
-
+                self.models3D[str(group)] = model
             elif modeltype == "2D":
-                ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 100, 'TeV')
+                ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 15, 'TeV')
                 offset = sqrt_space(start=0, stop=2.5, num=100) * u.deg
-                model = EnergyOffsetBackgroundModel(energy=ebounds, offset=offset)
+                model = EnergyOffsetBackgroundModel(ebounds, offset)
                 model.fill_obs(obs_ids=obs_ids, data_store=self.data_store, excluded_sources=self.excluded_sources)
                 model.compute_rate()
-                self.models2D.append(model)
+                self.models2D[str(group)] = model
             else:
                 raise ValueError("Invalid model type: {}".format(modeltype))
 
+    def filename(self, modeltype, group_id):
+        """Return the filename where is stored the background model for a certain groupid in zenith angle and efficiency
```

Comment:
> Return the filename where is stored the background model for a certain groupid in zenith angle and efficiency

Change to

```
"""Filename for a given ``modeltype`` and ``group_id``.
```

I think you're mentioning zenith angle and efficiency in other places where it's not appropriate.
Groups could be based on other parameters and all that matters here is the group ID.